### PR TITLE
Change to new C++ based addon interface style

### PR DESCRIPTION
**Pull request by AlwinEsch**
This is related to https://github.com/xbmc/xbmc/pull/12356

### CODE REVIEW

**Comment by AlwinEsch** [comment]
I think the unrar lib need a update in near future. Has seen during test that some rar files created by newer versions does not work with the addon here.

**Comment by notspiff** [comment]
Known. However it works for the target - stored archives. Lib has changed a
lot and my hacks would need complete rewrites.

**Comment by AlwinEsch** [comment]
Maybe I look for them in the next time.

File: src/RarFile.cpp [hunk 1]
```diff
@@ -492,14 +456,14 @@ ssize_t Read(void* context, void* lpBuf, size_t uiBufSize)
...
       // invalid data returned by UnrarXLib, prevent a crash
-      XBMC->Log(ADDON::LOG_ERROR, "CRarFile::Read - Data buffer in inconsistent state");
+      kodi::Log(ADDON_LOG_ERROR, "CRarFile::Read - Data buffer in inconsistent state");
       ctx->inbuffer = 0;
     }
 
     if (ctx->inbuffer == 0)
       break;
 
-    ssize_t copy = std::min(static_cast<ssize_t>(ctx->inbuffer), uicBufSize);
```
Comment (asavah) [inline src/RarFile.cpp]: The static_cast seems to be required, at least while cross compiling for ARM (rpi) with gcc 7.1.x
```
vfs.rar/src/RarFile.cpp: In member function ‘virtual ssize_t CRARFile::Read(void*, void*, size_t)’:
vfs.rar/src/RarFile.cpp:466:54: error: no matching function for call to ‘min(int64_t&, ssize_t&)’
     ssize_t copy = std::min(ctx->inbuffer, uicBufSize);
                                                      ^
```

https://stackoverflow.com/questions/14508022/error-no-matching-function-for-call-to-minlong-unsigned-int-unsigned-int

Comment (AlwinEsch) [inline src/RarFile.cpp]: Thanks, becomes changed back.

**Comment by AlwinEsch** [comment]
Is updated and part fixed.

**Pull request merged**